app/redux/Home/HomeSlice.tsx

```diff
@@ -0,0 +1,66 @@
+import {
+  ActionReducerMapBuilder,
+  createSlice,
+  Draft,
+  PayloadAction,
+} from '@reduxjs/toolkit';
+import { ApiResponseType, ApiErrorType } from '../../services';
+import { getTrendingData } from './HomeActions';
+import type { InitialStateType, ResetParams } from './HomeTypes';
+import type { RootState } from '../index';
+
+const initialState: InitialStateType = {
+  isLoading: false,
+  data: {},
+  error: null,
+};
+
+export const homeSlice = createSlice({
+  name: 'home',
+  initialState,
+  reducers: {
+    reset: (state: ResetParams) => {
```

Comment:
through all slice state types same not different everywhere so just simple to define the type of state like a state: Draft<InitialStateType> instead of state: ResetParams.